libexec/alluxio-config.sh

```diff
@@ -22,21 +22,67 @@ this="$config_bin/$script"
 # their own alluxio-layout.sh file to set system installation locations.
 if [[ -z "$ALLUXIO_SYSTEM_INSTALLATION" ]]; then
   VERSION=1.1.0-SNAPSHOT
-  export ALLUXIO_PREFIX=$(dirname $(dirname "$this"))
-  export ALLUXIO_HOME=${ALLUXIO_PREFIX}
-  export ALLUXIO_CONF_DIR="$ALLUXIO_HOME/conf"
-  export ALLUXIO_LOGS_DIR="$ALLUXIO_HOME/logs"
-  export ALLUXIO_JARS="$ALLUXIO_HOME/assembly/target/alluxio-assemblies-${VERSION}-jar-with-dependencies.jar"
+  ALLUXIO_HOME=$(dirname $(dirname "${this}"))
+  ALLUXIO_CONF_DIR="${ALLUXIO_HOME}/conf"
+  ALLUXIO_LOGS_DIR="${ALLUXIO_HOME}/logs"
+  ALLUXIO_JARS="${ALLUXIO_HOME}/assembly/target/alluxio-assemblies-${VERSION}-jar-with-dependencies.jar"
 fi
 
-# Make sure alluxio-env.sh exists
-if [[ ! -e $ALLUXIO_CONF_DIR/alluxio-env.sh ]]; then
-  echo "Cannot find alluxio-env.sh in $ALLUXIO_CONF_DIR."
-  echo "Please create one manually or using '$ALLUXIO_HOME/bin/alluxio bootstrap-conf'."
-  exit 1
+JAVA_HOME=${JAVA_HOME:-"$(dirname $(which java))/../.."}
+JAVA=${JAVA:-"${JAVA_HOME}/bin/java"}
+
+[ -f ${ALLUXIO_CONF_DIR}/alluxio-env.sh ] && . "${ALLUXIO_CONF_DIR}/alluxio-env.sh"
+
+if [[ -z "${ALLUXIO_MASTER_ADDRESS}" ]]; then
+  # Make sure alluxio-site.properties exists
+  if [[ ! -e $ALLUXIO_CONF_DIR/alluxio-site.properties ]]; then
+    echo "Cannot find alluxio-site.properties in ${ALLUXIO_CONF_DIR}."
+    echo "Please create one manually or using '${ALLUXIO_HOME}/bin/alluxio bootstrap-conf'."
+    exit 1
+  fi
 fi
 
-. $ALLUXIO_CONF_DIR/alluxio-env.sh
+if [[ -n "${ALLUXIO_HOME}" ]]; then
```

Comment:
but still, since in theory this var maybe wiped off in `alluxio-env.sh` and the source code has its defalut value, if we hardcode `-Dalluxio.home=${ALLUXIO_HOME}"` in `ALLUXIO_JAVA_OPTS`, I feel it is less optimal as this may overwrite the system default value for `alluxio.home`